pkg/commands/test/output_test.go

```diff
@@ -153,13 +154,57 @@ func Test_jsonOutputManager_put(t *testing.T) {
 		"Successes": []
 	}
 ]
+`,
+		},
+		{
+			msg: "handles structured errors",
+			args: args{
+				fileName: "-",
+				cr: CheckResult{
+					Failures: []error{StructuredError{"msg": "structured failure", "code": "ERR0001"}},
+				},
+			},
+			exp: `[
+	{
+		"filename": "",
+		"Warnings": [],
+		"Failures": [
+			"structured failure"
+		],
+		"Successes": []
+	}
+]
+`,
+		},
+		{
+			msg: "handles structured errors with details",
+			details: true,
+			args: args{
+				fileName: "-",
+				cr: CheckResult{
+					Failures: []error{StructuredError{"msg": "structured failure", "code": "ERR0001"}},
+				},
+			},
+			exp: `[
+	{
+		"filename": "",
+		"Warnings": [],
+		"Failures": [
+			{
+				"code": "ERR0001",
+				"msg": "structured failure"
+			}
+		],
+		"Successes": []
+	}
+]
```

Comment:
This test is rather fragile as Go randomizes the order of maps, I'm not sure if that also applies to when it serializes to json or if that is (semi-)well-ordered...

Reply:
I _think_ the [current implementation](https://golang.org/src/encoding/json/encode.go#L629) is deterministic so this should be ok!